src/local-settings.js

```diff
@@ -2,6 +2,10 @@ const DEFAULT_ETH_NODE_KEY = 'ETH_NODE_KEY'
 const ENS_REGISTRY_ADDRESS_KEY = 'ENS_REGISTRY_ADDRESS_KEY'
 const ETH_NETWORK_TYPE_KEY = 'ETH_NETWORK_TYPE_KEY'
 const IPFS_GATEWAY_KEY = 'IPFS_GATEWAY_KEY'
+const IPFS_RPC_KEY = 'IPFS_RPC_KEY'
+
+// (protocol)://(host):(port)
+const IPFS_RPC_REGEX = /^([\w]*):\/\/(.*):(\d*)$/
```

Comment:
This is a fairly naive regex, but should be fine for our needs. I assume if someone sets something that passes this, but doesn't work for `js-ipfs-api`, that'll complain (e.g. can't connect to node, weird protocol, etc).

Reply:
Not super important because we rely on `js-ipfs-api`, but we should probably use `+` instead of `*` for the protocol, host and port as they are not optional?

Or maybe we could use [`new URL(str)`](https://developer.mozilla.org/en-US/docs/Web/API/URL)?

It’s available in the browsers we support: https://caniuse.com/#feat=url